File: Python3/Easy/WrongAnswer/find-subarrays-with-equal-sum_1.py
class Solution:
    def findSubarrays(self, nums: List[int]) -> bool:
        
        # O(n^2) - 2 nested for loops
        # O(1) - constant space
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if sum(nums[i:j]) == sum(nums[j:j + 2]):
                    return True
        return False

File: C++/Easy/WrongAnswer/find-subarrays-with-equal-sum_1.cpp
class Solution {
public:
    bool findSubarrays(vector<int>& nums) {
        
        int n = nums.size();
        for (int i = 0; i < n - 1; i++) {
            int sum = nums[i];
            for (int j = i + 1; j < n; j++) {
                sum += nums[j];
                if (sum == 0) {
                    return true;
                }
            }
        }
        return false;
    }
};